Nc-News/src/components/Article.jsx

```diff
@@ -1,25 +1,75 @@
 import { useEffect, useState } from "react";
 import { Link, useParams } from "react-router-dom";
-import { getArticle, getComments } from "../api";
+import { getArticle, getComments, patchArticle } from "../api";
 import { CommentCard } from "./Containers/CommentCard";
+import UpEgg from "./images/UpEgg.svg?react";
+import DownEgg from "./images/DownEgg.svg?react";
+import GreyDownEgg from "./images/GreyDownEgg.svg?react";
+import GreyUpEgg from "./images/GreyUpEgg.svg?react";
 
 export function Article() {
   const { article_id } = useParams();
   const [article, setArticle] = useState({});
   const created = new Date(article.created_at);
   const [isLoading, setIsLoading] = useState(true);
+
   const [comments, setComments] = useState([]);
+  const [vote, setVote] = useState(0);
+  const [votes, setVotes] = useState(0);
+
+  const resetVote = () => {
+    if (vote >= 1) {
+      setVotes((currentVotes) => {
+        return votes - 1;
+      });
+      patchArticle(article.article_id, -1);
+    } else if (vote <= -1) {
+      setVotes((currentVotes) => {
+        return votes + 1;
+      });
+      patchArticle(article.article_id, 1);
+    }
+    setVote(0);
+  };
+
+  const upVote = () => {
+    if (vote <= -1) {
+      setVote(2);
+    } else if (vote === 0) {
+      setVote(1);
+    }
+  };
+  const DownVote = () => {
+    if (vote >= 1) {
+      setVote(-2);
+    } else if (vote === 0) {
+      setVote(-1);
+    }
+  };
 
   useEffect(() => {
     setIsLoading(true);
     Promise.all([getArticle(article_id), getComments(article_id)]).then(
       ([article, comments]) => {
         setArticle(article.data.article);
         setComments(comments.data.comments);
+        setVotes(article.data.article.votes);
         setIsLoading(false);
       }
     );
   }, []);
+
+  useEffect(() => {
+    setVotes((currentVotes) => {
+      console.log(votes);
+      return votes + vote;
+    });
+  }, [vote]);
+
```

Comment:
If you click the upvote or downvote eggs a few times in a row, the state stops reflecting the real value of the votes 🔍 🥚 
A bug to have a look at 🐛